sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,153 @@
+        public PathAccessControlEntry() { }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="accessControlType">Specifies which role this entry targets.</param>
+        /// <param name="permissions">Specifies the permissions granted to this entry.</param>
+        /// <param name="defaultScope">Indicates whether this is the default entry for the ACL.</param>
+        /// <param name="entityId">Optional entity ID to which this entry applies.</param>
+        public PathAccessControlEntry(
+            AccessControlType accessControlType,
+            RolePermissions permissions,
+            bool defaultScope = false,
+            string entityId = default)
+        {
+            if (entityId != null
+                && !(accessControlType == AccessControlType.User || accessControlType == AccessControlType.Group))
+            {
+                throw new ArgumentException("AccessControlType must be User or Group if entityId is specified.");
+            }
+
+            DefaultScope = defaultScope;
+            AccessControlType = accessControlType;
+            EntityId = entityId;
+            Permissions = permissions;
+        }
+
+        /// <summary>
+        /// Override of ToString().
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            if (DefaultScope)
+            {
+                stringBuilder.Append("default:");
+            }
+            stringBuilder.Append(AccessControlType.ToString().ToLowerInvariant());
+            stringBuilder.Append(":");
+            stringBuilder.Append(EntityId ?? "");
+            stringBuilder.Append(":");
+            stringBuilder.Append(Permissions.ToSymbolicRolePermissions());
+
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Parses the provided string into a <see cref="PathAccessControlEntry"/>
+        /// </summary>
+        /// <param name="s">The string representation of the access control list.</param>
+        /// <returns>A <see cref="PathAccessControlEntry"/>.</returns>
+        public static PathAccessControlEntry Parse(string s)
+        {
+            if (s == null)
+            {
+                return null;
+            }
+
+            PathAccessControlEntry entry = new PathAccessControlEntry();
+            string[] parts = s.Split(':');
+            int indexOffset = 0;
+
+            if (parts.Length < 3 || parts.Length > 4)
+            {
+                throw new ArgumentException($"{nameof(s)} should have 3 or 4 parts delimited by colons");
+            }
+
+            if (parts.Length == 4)
+            {
+                if (!parts[0].Equals("default", StringComparison.OrdinalIgnoreCase))
+                {
+                    throw new ArgumentException($"If {nameof(s)} is 4 parts, the first must be \"default\"");
```

Comment:
Move to DataLakeErrors.cs and include the actual string you're trying to parse

Reply:
Added actual string, see my comment above.

Reply:
foxed/